contracts/strategies/morpho/MetaMorphoStrategy.sol

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.10;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import "../../interfaces/IERC20Detailed.sol";
+import "../../interfaces/morpho/IMorphoSupplyVault.sol";
+import "../../interfaces/morpho/IMorphoAaveV2Lens.sol";
+import "../../interfaces/morpho/IMorphoCompoundLens.sol";
+import "../../interfaces/morpho/IRewardsDistributor.sol";
+import "../../interfaces/morpho/IURD.sol";
+import "../../interfaces/morpho/IMetamorphoSnippets.sol";
+import "../ERC4626Strategy.sol";
+
+contract MetaMorphoStrategy is ERC4626Strategy {
+  using SafeERC20Upgradeable for IERC20Detailed;
+
+  //// @notice MORPHO governance token
+  IERC20Detailed internal constant MORPHO = IERC20Detailed(0x9994E35Db50125E0DF82e4c2dde62496CE330999);
+
+  /// @notice reward token address (e.g. MORPHO, COMP, ...)
+  address[] public rewardTokens;
+
+  /// @notice MORPHO transferability
+  bool public morphoTransferable;
+
+  /// @notice Metamorpho snippets contract, used for apr
+  address public mmSnippets;
+
+  /// Initialization
+
+  /// @notice initialize the strategy
+  /// @param _strategyToken strategy token address
+  /// @param _token underlying token address
+  /// @param _owner owner address
+  /// @param _mmSnippets snippets contract address
+  /// @param _rewardTokens array of reward tokens
+  function initialize(
+    address _strategyToken,
+    address _token,
+    address _owner,
+    address _mmSnippets,
+    address[] memory _rewardTokens
+  ) public {
+    _initialize(_strategyToken, _token, _owner);
+
+    // used to fetch apr
+    mmSnippets = _mmSnippets;
```

Comment:
the logic could be integrated here maybe?

Reply:
wdym exactly here?

Reply:
I mean copy past the snippet logic in this contract. But nvm this is just a suggestion

Reply:
It's a bit too much code to include everything in the strategy directly we prefer relying on the mm snippets contract directly